core/src/main/java/io/questdb/cairo/TxWriter.java

```diff
@@ -164,9 +164,13 @@ public void commit(int commitMode, ObjList<? extends SymbolCountProvider> symbol
 
             prevRecordBaseOffset = lastRecordBaseOffset;
             lastRecordBaseOffset = writeBaseOffset;
+            int commitMode = configuration.getCommitMode();
+            if (commitMode != CommitMode.NOSYNC) {
+                txMemBase.sync(commitMode == CommitMode.ASYNC);
```

Comment:
We should always sync metadata files if we want better stability on power loss.

Reply:
unfortunately when we have NOSYNC committing txn only is like doing a CPR to a dead person

Reply:
i guess I need to clarify. If someone chose not to SYNC at all, my assumption would be they did that for performance reasons understanding and accepting the risks. Syncing TXN could increase commit latency making small commits significantly slower

Reply:
If we aim for stability in the face of a power loss even with nosync, we should msync/fsync all metadata files. This PR adds a bunch of non-conditional msync/fsync calls for various metadata files, but not this one, so the overall logic seems inconsistent.

Also, it's not uncommon to expose a setting for syncing data files, but having always-on, non-configurable synchronization for metadata files - CH is an example of such an approach.

Reply:
we actually have a benchmark for this, i will post results here